synapse/storage/state.py

```diff
@@ -361,18 +361,20 @@ def approx_difference(self, subtrahend: "StateFilter") -> "StateFilter":
         Returns a state filter which represents `self - subtrahend`.
 
         The resultant state filter MUST admit all state events that are admitted
```

Comment:
I think we probably want something in prose that explicitly describes the `⊆ E(self)` bit. How about this?

```suggestion
The resultant filter

- MUST admit all state events that are admitted by only this filter (`self`) and not `subtrahend`;
- MUST NOT admit state events rejected by this filter (`self`); and
- MAY be an over-approximation: it may additionally admit some state events from `subtrahend`.
```